frontend/src/embedding-sdk-bundle/components/public/dashboard/use-common-dashboard-params.tsx

```diff
@@ -127,19 +131,23 @@ const findDashboardById = (
   dashboardId: DashboardId,
   dashboards: Record<DashboardId, StoreDashboard>,
 ): StoreDashboard | null => {
-  if (typeof dashboardId === "number" || isJWT(dashboardId)) {
-    return dashboards[dashboardId] ?? null;
-  }
-
   // Lookup via entity ids.
   // Dashboards are a mapping of numeric id to dashboard.
-  if (typeof dashboardId === "string") {
+  if (isBaseEntityID(dashboardId)) {
     return (
       Object.values(dashboards).find(
         (dashboard) => dashboard.entity_id === dashboardId,
       ) ?? null
     );
   }
 
+  if (isNumericStringOrNumber(dashboardId) || isJWT(dashboardId)) {
+    return dashboards[dashboardId] ?? null;
+  }
+
   return null;
 };
+
+function isNumericStringOrNumber(value: string | number): boolean {
```

Comment:
This might incorrectly return 0 `true` for empty strings, could that be an issue?

Reply:
I'll sort this out 🙏